packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts

```diff
@@ -120,6 +128,70 @@ export class InsuredBridgeL1Client {
     return this.whitelistedTokens[chainId];
   }
 
+  getRateModelForBlockNumber(l1Token: string, blockNumber: number | undefined = undefined): RateModel {
+    this._throwIfNotInitialized();
+    const l1TokenNormalized = toChecksumAddress(l1Token);
+
+    if (!this.rateModelsForToken[l1TokenNormalized])
+      throw new Error(`No updated rate model events for L1 token: ${l1TokenNormalized}`);
+
+    let rateModelToReturn: any;
+    // Rate model events are inserted into the array from oldest at index 0 to newest at index length-1, so we'll
```

Comment:
I wasn't sure whether this logic of parsing the rate model and checking that its in the expected shape should go in this function or the `update` function